hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AutoAccountCreator.java

```diff
@@ -0,0 +1,239 @@
+        // TODO: Not sure if fee should be set here childRecord.transactionFee(fee);
+    }
+
+    /**
+     * Get fees for finalization of lazy creation.
+     * @return fee for finalization of lazy creation
+     */
+    private long getLazyCreationFinalizationFee() {
+        final var updateTxnBody = CryptoUpdateTransactionBody.newBuilder()
+                .key(Key.newBuilder().ecdsaSecp256k1(Bytes.EMPTY).build());
+        return autoCreationFeeFor(TransactionBody.newBuilder().cryptoUpdateAccount(updateTxnBody));
+    }
+
+    /**
+     * Get fees for auto creation.
+     * @param syntheticCreation transaction body for auto creation
+     * @return fee for auto creation
+     */
+    private long autoCreationFeeFor(final TransactionBody.Builder syntheticCreation) {
+        final var topLevelPayer = handleContext.body().transactionIDOrThrow().accountIDOrThrow();
+        final var payerAccount = accountStore.get(topLevelPayer);
+        validateTrue(payerAccount != null, PAYER_ACCOUNT_NOT_FOUND);
+        final var txn = Transaction.newBuilder().body(syntheticCreation.build()).build();
+        //        final var fees = handleContext.feeCalculator().computePayment(txn, payerAccount.key());
+        //        return fees.serviceFee() + fees.networkFee() + fees.nodeFee();
+        // TODO : need to use fee calculator
+        return 100;
+    }
+
+    /**
+     * Create a transaction body for new hollow-account with the given alias.
+     * @param alias alias of the account
+     * @param balance initial balance of the account
+     * @return transaction body for new hollow-account
+     */
+    public TransactionBody.Builder createHollowAccount(final Bytes alias, final long balance) {
+        final var baseBuilder = createAccountBase(balance);
+        baseBuilder.key(IMMUTABILITY_SENTINEL_KEY).alias(alias).memo(LAZY_MEMO);
+        return TransactionBody.newBuilder().cryptoCreateAccount(baseBuilder.build());
+    }
+
+    /**
+     * Create a transaction body for new account with the given balance and other common fields.
+     * @param balance initial balance of the account
+     * @return transaction body for new account
+     */
+    private CryptoCreateTransactionBody.Builder createAccountBase(final long balance) {
+        return CryptoCreateTransactionBody.newBuilder()
+                .initialBalance(balance)
+                .autoRenewPeriod(Duration.newBuilder().seconds(THREE_MONTHS_IN_SECONDS));
+    }
+
+    /**
+     * Create a transaction body for new account with the given alias, key, balance and maxAutoAssociations.
+     * @param alias alias of the account
+     * @param key key of the account
+     * @param balance initial balance of the account
+     * @param maxAutoAssociations maxAutoAssociations of the account
+     * @return transaction body for new account
+     */
+    private TransactionBody.Builder createAccount(
+            final Bytes alias, final Key key, final long balance, final int maxAutoAssociations) {
```

Comment:
Is this `maxAutoAssociations` as in only X auto associations created, or is this the number of auto associations that need to be created? A little confusing

Reply:
Those are the maxAutoAssociation slots to be set on the auto- created account

Reply:
Ahh got it, makes sense